pyccel/ast/operators.py

```diff
@@ -781,6 +779,12 @@ class PyccelEq(PyccelComparisonOperator):
     """
     __slots__ = ()
 
+    def __new__(cls, arg1, arg2, simplify = False):
```

Comment:
I don't really understand the use of the function parameter simplify for both `PyccelEq` and `PyccelNe`, would you mind explain it to me ?

Reply:
This argument is inherited from the class `PyccelBinaryOperator`. It is used when we use pyccel to add operations (usually mathematical operations) and we don't want to print them fully.
E.g. if we have:
```
for i in range(5):
    a = 1+2
```
the operator `+` should be printed fully as it appears in the original code.
But in fortran a loop is printed with inclusive bounds. `do i = 0_i64, 5_i64 - 1_i64` is a bit unnatural, we prefer to print the more natural `do i = 0_i64, 4_i64` so here we don't want it printed fully. Instead we use `simplify = True` when creating this `PyccelMinus` operator to tell the class that it is allowed to make simplifications (e.g. literal operations simplify to their value, `a+0` simplifies to `a`, etc)

Reply:
Oh i understand, thank for this clarification !